### PR TITLE
chore: Add openai response tests

### PR DESCRIPTION
**Pull request by deekshas8**
## Context

AI/gen-ai-hub-sdk-js-backlog#143.

Add more test. I also noticed a bug in the convenience method in embedding response class. 

## Definition of Done

- [x] Code is tested (Unit, E2E)
- [ ] Error handling created / updated & covered by the tests above
- [ ] Documentation updated
- [ ] (Optional) Aligned changes with the Java SDK
- [ ] (Optional) Release notes updated

### CODE REVIEW

File: packages/foundation-models/src/azure-openai/azure-openai-embedding-response.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export class AzureOpenAiEmbeddingResponse {
    */
   getEmbedding(dataIndex = 0): number[] | undefined {
     this.logInvalidDataIndex(dataIndex);
-    return this.data.data[0]?.embedding;
+    return this.data.data[dataIndex]?.embedding;
```
Comment (tomfrenken) [inline packages/foundation-models/src/azure-openai/azure-openai-embedding-response.ts:30]: lmao

Comment (ZhongpinWang) [reply packages/foundation-models/src/azure-openai/azure-openai-embedding-response.ts:30]: 🧯🔥

File: packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts [hunk 2]
```diff
@@ -14,9 +15,42 @@ describe('OpenAI chat completion response', () => {
...
+  });
 
-  it('should return the completion response', () => {
-    expect(response.data).toStrictEqual(mockResponse);
+  it('should return undefined when convenience function is called with incorrect index', () => {
+    const logger = createLogger({
+      package: 'foundation-models',
+      messageContext: 'azure-openai-chat-completion-response'
+    });
+    const errorSpy = jest.spyOn(logger, 'error');
```
Comment (ZhongpinWang) [inline packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts:50]: [q] Can we spy on the private method `logInvalidDataIndex` directly?

```ts
const spy = jest.spyOn(azureOpenAiChatResponse as any, 'logInvalidDataIndex');
expect(spy).toHaveBeenCalledWith(1);
expect(spy).toHaveBeenCalledTimes(1);
```

File: packages/foundation-models/src/azure-openai/azure-openai-embedding-response.test.ts [hunk 3]
```diff
@@ -12,9 +14,23 @@ describe('Azure OpenAI embedding response', () => {
...
+  });
+  it('should return undefined when convenience function is called with incorrect index', () => {
+    const logger = createLogger({
+      package: 'foundation-models',
+      messageContext: 'azure-openai-embedding-response'
+    });
+    const errorSpy = jest.spyOn(logger, 'error');
+    expect(embeddingResponse.getEmbedding(1)).toBeUndefined();
+    expect(errorSpy).toHaveBeenCalledWith('Data index 1 is out of bounds.');
+    expect(errorSpy).toHaveBeenCalledTimes(1);
```
Comment (ZhongpinWang) [inline packages/foundation-models/src/azure-openai/azure-openai-embedding-response.test.ts:34]: Same here.

File: packages/orchestration/src/orchestration-response.test.ts [hunk 4]
```diff
@@ -39,8 +41,15 @@ describe('OrchestrationResponse', () => {
...
   it('should return undefined when convenience function is called with incorrect index', () => {
+    const logger = createLogger({
+      package: 'orchestration',
+      messageContext: 'orchestration-response'
+    });
+    const errorSpy = jest.spyOn(logger, 'error');
     expect(orchestrationResponse.getFinishReason(1)).toBeUndefined();
+    expect(errorSpy).toHaveBeenCalledWith('Choice index 1 is out of bounds.');
     expect(orchestrationResponse.getContent(1)).toBeUndefined();
+    expect(errorSpy).toHaveBeenCalledTimes(2);
```
Comment (ZhongpinWang) [inline packages/orchestration/src/orchestration-response.test.ts:52]: Same here

File: packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts (hunk 2 above)
Comment (tomfrenken) [reply packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts:50]: While possible,  I don't know if hacking our way around typescript is the best practice here.
On the top of my mind, I don't see a strong reason opposing this approach though.

**Review by tomfrenken — APPROVED** [review]
LGTM, consider if you prefer the approach of @ZhongpinWang though

File: packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts (hunk 2 above)
Comment (deekshas8) [reply packages/foundation-models/src/azure-openai/azure-openai-chat-completion-response.test.ts:50]: The thing is this function will always be called, but doesn't mean the logger will also be. I want to test that the logger.error is called.

**Pull request merged**